opts/opts.go

```diff
@@ -182,27 +212,33 @@ func ValidateDevice(val string) (string, error) {
 //    [host-dir:]container-path[:rw|ro]
 // It will also validate the mount mode.
 func ValidatePath(val string) (string, error) {
-	return validatePath(val, true)
+	return validatePath(val, volumeMode)
 }
 
-func validatePath(val string, validateMountMode bool) (string, error) {
+func validatePath(val string, validateMode int) (string, error) {
```

Comment:
I tend to agree with @calavera, it would be better passing the validator function. 

> Because they don't have the same return value requirements for different validation functions, and it might be more flexible if we just pass an int value for validate type and let the function decide what to do, maybe it'll be simple enough to not call a validate function. WDYT?

We are already doing that for other validator function (with this `ValidatePath` function — [ValidatorFctType](https://github.com/docker/docker/blob/master/opts/opts.go#L149)) ; The `validateDeviceMode` would return `bool, nil` for all cases — that's a small tradeoff for a cool looking method :sweat_smile:.

But like @calavera this is also my personal opinion.